api/lambda/authorizer.js

```diff
@@ -315,70 +318,128 @@ AuthPolicy.prototype = (function AuthPolicyClass() {
   };
 }());
 
-exports.Custom = (event, context, callback) => {
-  //console.log('Client authorization token:', event.authorizationToken);
-  //console.log('Method ARN:', event.methodArn);
-
-  // TODO: Move this initial key lookup step into constructor for on-time run only.
-  // Download and store the JWT set for our Cognito User Pool
-  // let jwtKeySet = 'https://cognito-idp.' + config.AWS_REGION + '.amazonaws.com/' + env.config.USER_POOL_ID + '/.well-known/jwks.json';
-  // logger.info('jwtKeySetUrl:', jwtKeySet);
-
-
-  // Decode and validate the incoming token
-  var decoded = null;
-  try {
-    decoded = jwt.decode(event.authorizationToken);
-    logger.info('JWT Decoded', decoded);
-  } catch(e) {
-    callback(e);
+
+function processAuthRequest(event, tokenIssuer, callback) {
+  var token = event.authorizationToken;
+
+  //Fail if the token is not jwt
+  var decodedJwt = jwt.decode(token, {complete: true});
+  if (!decodedJwt) {
+    console.log('Not a valid JWT token');
+    callback('Unauthorized');
```

Comment:
Instead of returning the error "Unauthorized," in any non-happy path, the callback here should instead return a well-formed IAM policy that denies everything. You can use a helper function to build this and the built-in denyAllMethods operation to build the JSON object to return.